File: frontend/my-app/src/app/models/Note.ts
export class Note{
    private name: string;
    private description: string;

    constructor(name:string, description:string) {
        this.name = name;
        this.description = description;
    }

        get _name():string{
            return this._name;
        }

        set _name(newName:string){
             this._name = newName;
        }


        get _description():string{
            return this.description;
        }
    
        set _description(newName:string){
            this.description = newName;
       }

       
}

File: task-Management/src/main/java/com/example/taskManagement/model/Note.java
package com.example.taskManagement.model;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Getter
@Setter
// @Entity(name = "NOTE")
@Document(collection = "collection")
public class Note {

    // @GeneratedValue(strategy = GenerationType.IDENTITY) // this relies on a
    // database to auto-increment and generate and id

    // private static int nextId = 1; // Static field to track the next available ID

    // @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private String id;

    // private int noteNumber;
    private String name;
    private String description;
    private boolean completed;
    private LocalDateTime deadline;
    private ArrayList<Tag> tags;
    private String category;
    private int priority;

    // below is used without a databased
    // public Note() {
    // this.id = nextId++; // Assign the next available ID and increment the counter
    // this.tags = new ArrayList<>();
    // }

}
